src/shared/efi-loader.c

```diff
@@ -67,26 +67,37 @@ assert_cc(sizeof(struct device_path) == sizeof(struct device_path__packed));
 
 int efi_reboot_to_firmware_supported(void) {
         _cleanup_free_ void *v = NULL;
+        static int cache = -1;
```

Comment:
I was wondering whether we should also make a cache in /run for those things. The idea would be to atomically move the cache in place, so that the first process that reads the variable creates the cache. But let's merge this PR first, and if the slowdowns are still an issue, we can pursue that.